core/src/test/scala/unit/kafka/admin/UserScramCredentialsCommandTest.scala

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.admin
+
+import java.io.{ByteArrayOutputStream, PrintStream}
+import java.nio.charset.StandardCharsets
+
+import kafka.server.BaseRequestTest
+import kafka.utils.Exit
+import org.junit.Assert._
+import org.junit.Test
+
+class UserScramCredentialsCommandTest extends BaseRequestTest {
+  override def brokerCount = 1
+  var exitStatus: Option[Int] = None
+  var exitMessage: Option[String] = None
+
+  case class ConfigCommandResult(stdout: String, exitStatus: Option[Int] = None)
+
+  private def runConfigCommandViaBroker(args: Array[String]) : ConfigCommandResult = {
+    val byteArrayOutputStream = new ByteArrayOutputStream()
+    val utf8 = StandardCharsets.UTF_8.name
+    val printStream = new PrintStream(byteArrayOutputStream, true, utf8)
+    var exitStatus: Option[Int] = None
+    Exit.setExitProcedure { (status, _) =>
```

Comment:
How does this interact with JUnit running tests concurrently?  cc @ijuma @hachikuji

Reply:
Hmm, good point, I think there may be a problem here in general because there is only a single exit procedure that can be set globally, and multiple tests that set/reset it in parallel will collide.  There are 16 Scala test classes in `core` out of 260 that do this -- so 4% of test classes.  So I think this will introduce some flakiness to these 16 tests.  Does this sound correct to you, and should we open a separate ticket for this as opposed to trying to fix it here?

Reply:
Actually, I think this may not be an issue since parallel tests in Gradle run in separate processes rather than separate threads.  From https://docs.gradle.org/current/dsl/org.gradle.api.tasks.testing.Test.html: `"Test are always run in (one or more) separate JVMs."`